runtime/JavaScript/package.json

```diff
@@ -20,27 +34,29 @@
   },
   "homepage": "https://github.com/antlr/antlr4",
   "devDependencies": {
-    "@babel/core": "^7.19.1",
-    "@babel/eslint-parser": "^7.19.1",
-    "@babel/preset-env": "^7.19.4",
-    "@types/node": "^18.7.23",
-    "babel-loader": "^8.2.5",
+    "@babel/cli": "^7.21.0",
+    "@babel/core": "^7.21.3",
+    "@babel/eslint-parser": "^7.21.3",
+    "@babel/preset-env": "^7.20.2",
+    "@rollup/plugin-babel": "^6.0.3",
+    "@types/node": "^18.15.3",
     "c8": "^7.12.0",
-    "compression-webpack-plugin": "^10.0.0",
```

Comment:
so what happens to compression ?

Reply:
compression doesn't really make sense here. Webpack is used for shipping web applications primarily, and this project is a library... In any case, this only impacts the umd bundle, which nobody really uses.

Reply:
why build umd if 'nobody really uses it' ? And what makes you think that antlr is not used in web apps ? I do.

Reply:
we can remove the umd if you'd like. I basically only left it there because it was there before and I was trying to minimize the changes in this PR.

I created this whole PR since we are using antlr4 on the web, so trust me I know how it's used, and this PR is designed specifically to make antlr4 less painful to consume in a web application by correctly setting up the `type`, `main`, `module` & `exports` fields.

Reply:
there was no umd before, so not sure I follow...

Reply:
the bundle you used to create with webpack was also in umd format

Reply:
internally, but exposed as esm

Reply:
I'm referring to [this umd formatted build output](https://unpkg.com/browse/antlr4@4.12.0/dist/antlr4.web.js) that was pointed to by the [package.json](https://unpkg.com/browse/antlr4@4.12.0/package.json) `browser` field.